arbor/include/arbor/morph/label_dict.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+
+#include <arbor/morph/locset.hpp>
+#include <arbor/morph/region.hpp>
+#include <arbor/util/optional.hpp>
+
+namespace arb {
+
+class label_dict {
+    using ps_map = std::unordered_map<std::string, arb::locset>;
+    using reg_map = std::unordered_map<std::string, arb::region>;
+    ps_map locsets_;
+    reg_map regions_;
+
+public:
+    label_dict();
+    label_dict(const label_dict&);
+    label_dict(label_dict&&);
+
+    void set(const std::string& name, locset p);
+    void set(const std::string& name, region r);
+
+    util::optional<arb::region&>       region(const std::string& name);
+    util::optional<const arb::region&> region(const std::string& name) const;
```

Comment:
It is my error that the `set` method silently drops insertion on an (ignorance on my part regarding how `std::unordered_map::instert` behaves when inserting with a key that is already in the table).

I agree with your proposed changes (because you are right, it is icky design).

Reply:
Fixed.